packages/arb-token-bridge-ui/src/components/TransferPanel/hooks/useTransferRequiresChainSwitch.ts

```diff
@@ -0,0 +1,36 @@
+import { useLatest } from 'react-use'
+
+import { useNetworksRelationship } from '../../../hooks/useNetworksRelationship'
+import { isNetwork } from '../../../util/networks'
+import { useNetworks } from '../../../hooks/useNetworks'
+import { useIsConnectedToOrbitChain } from '../../../hooks/useIsConnectedToOrbitChain'
+import { useIsConnectedToArbitrum } from '../../../hooks/useIsConnectedToArbitrum'
+
+export function useTransferRequiresChainSwitch() {
```

Comment:
what if this was a return value from `useTransferReadiness`?

Reply:
i deleted the hook and used a simple condition `const isConnectedToTheWrongChain =
      chainId !== latestNetworks.current.sourceChain.id` instead

also abstracted the relevant code from `transferCctp` and `transfer` into a reusable function